hayeon/baekjoon/문자열/15829-Hashing.js

```diff
@@ -0,0 +1,80 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const [num, strings] = input;
+  const mod = 1234567891;
+  let r = 1;
+  let result = 0;
+
+  for (let i = 0; i < +num; i++) {
+    // result += Math.pow(31, strings.charCodeAt(i) - 96);
+    result += (strings.charCodeAt(i) - 96) * r;
+    result %= mod;
+
+    r *= 31;
+    r %= mod;
+  }
+
+  console.log(result);
```

Comment:
```suggestion
  const [num, strings] = input;
  const mod = 1234567891;
  let r = 1;
  let result = 0;

  for (let i = 0; i < +num; i++) {
    result += (strings.charCodeAt(i) - 96) * r;

    r *= 31;
    r %= mod;
  }

  console.log(result % mod);
```
- 풀이 잘봤습니다 🙇‍♂️
- charCodeAt을 이용한 ASCII 코드로 접근한 풀이 좋았습니다 :) 
- 카톡에서 `**` 연산자 쓰면 안된다고 말씀드렸는데 Math.pow나 **나 둘다 결국 같은 로직이라 같은 결과였을 겁니다 
- result는 매번 mod도 나누지 않고 마지막 결과 출력 시에만 1회 mod로 나눠주셔도 같습니다.